monitoring/rid_qualifier/test_executor.py

```diff
@@ -0,0 +1,145 @@
+            with open(flight_track_path) as generated_rid_state:
+                rid_state_data = json.load(generated_rid_state)
+
+            
+            effective_after = rid_state_data['reference_time']
+            
+            operator_location = OperatorLocation(lat = utm_sp['operator_details'][utmsp_index]['location']['latitude'], lng = utm_sp['operator_details'][utmsp_index]['location']['longitude'])
+            operator_id = str(uuid.uuid4())
+            operator = Operator(id = operator_id, location = operator_location, operation_description=  utm_sp['flight_details'][utmsp_index]['operation_description'] , serial_number = utm_sp['flight_details'][utmsp_index]['serial_number'],  registration_number =  utm_sp['flight_details'][utmsp_index]['registration_number'])
+
+            rid_flight_details = RIDFlightDetails(operator_id = operator.id, operation_description = operator.operation_description, serial_number = operator.serial_number, registration_number = operator.registration_number)
+
+            test_flight_details = TestFlightDetails(effective_after= effective_after,details = rid_flight_details)
+            test_flight = TestFlight(injection_id = str(uuid.uuid4()), telemetry= rid_state_data['flight_telemetry'], details_respones= test_flight_details)            
+            test_flight_deserialized = self.rid_serializer.make_json_compatible(test_flight)
+            requested_flights.append(test_flight_deserialized)
+            test_payload = {'test_id': str(uuid.uuid4()), "requested_flights": requested_flights}        
+
+            all_test_payloads.append({'injection_url':utm_sp['injection_url'], 'injection_payload': test_payload})        
+        
+        return all_test_payloads
+
+
+class TestSubmitter():
+    ''' A class to submit the test data to the UTMSP end point '''
+
+    def __init__(self, test_payloads):
+        self.test_payload_valid(test_payloads)
+
+        self.submit_payload(test_payloads)
+
+
+    def test_payload_valid(self, test_payloads: List) -> None:
+        ''' This method checks if the test definition is a valid JSON ''' #TODO : Have a comprehensive way to check JSON definition
+        if len(test_payloads):
+            pass
+        else:
+            raise ValueError("A valid payload object with atleast one flight / UTMSP must be submitted")
+
+    def submit_payload(self, test_payloads: List) -> None:
+        ''' This method submits the payload to indvidual UTMSP '''
+        my_test_harness = TestHarness()
+        for payload in test_payloads: 
+            my_test_harness.submit_test(payload)
+        
+
+
+class RIDDataValidator():
+    ''' A class to check the output from UTMSP in Flight Blender and produce a text report and write on disk '''
+    pass
+
+
+if __name__ == '__main__':
+        
+    # This is the configuration for the test.
+    test_configuration = {
+        "locale": "che",
```

Comment:
It would be good to document this format as I think this will be the primary way an end user interacts with rid_qualifier.  In that documentation, I think it will be important to mention that "locale" is what selects which set of test data to use for each SP under test.

Reply:
OK I have now done the first pass at it.